tools/shared.py

```diff
@@ -374,7 +374,7 @@ def perform_sanity_checks():
 
 
 @ToolchainProfiler.profile()
-def check_sanity(force=False):
+def check_sanity():
```

Comment:
Why does this remove the "force" param? Now the emcc locations that used to force will no longer do so? Is the idea that since we no longer always lock, that's ok?

Reply:
I refactored to avoid the force paramater.  Locations that want to always do a check now do so with `perform_sanity_checks` directly (the inner function that actually does the checkes) rather than passing this extra param to the outer function

Reply:
Let me refactor again and revert that part of this change which is maybe not necessary.

Reply:
Done